src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java

```diff
@@ -327,6 +328,58 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
         .filter(tm -> tm.getOCDS().getId().equalsIgnoreCase(userId)).findFirst().orElseThrow();
   }
 
+  /**
+   * Get Projects
+   * @return Collection of projects
+   * @param principal
+   */
+  public Collection<ProjectPackageSummary> getProjects(final String principal) {
+    // Fetch Jaggaer ID and Buyer company ID from Jaggaer profile based on OIDC login id
+    var jaggaerUserId = userProfileService.resolveBuyerUserByEmail(principal)
+            .orElseThrow(() -> new AuthorisationFailureException("Jaggaer user not found")).getUserId();
+    ProjectListResponse projectListResponse = jaggaerService.getProjectList(jaggaerUserId);
+    if (!CollectionUtils.isEmpty(projectListResponse.getProjectList().getProject())) {
+      return projectListResponse.getProjectList().getProject().stream()
+              .map(this::convertProjectToProjectPackageSummary)
+              .filter(projectPackageSummary -> projectPackageSummary != null) // TODO any better?
+              .collect(Collectors.toList());
+    }
+    return Collections.EMPTY_LIST;
+  }
+
+  /**
+   * convert project top project package summary
+   * @param project the project
+   * @return ProjectPackageSummary
+   */
+  private ProjectPackageSummary convertProjectToProjectPackageSummary(Project project) {
+	  // Get Project from database
+	  var dbProject = retryableTendersDBDelegate
```

Comment:
For me, again not representing reality as I have loads of old projects in Jaggaer, this results in a lot od individual calls to the DB (one for each project in Jaggaer).  Suggest perhaps changing the query to `findByExternalProjectIdIn(Set<String> externalProjectIds)` - that should get a collection and then could use that to do the compare between what is in Jaggaer and what is in the DB perhaps?

Reply:
done